unable to configure on osx 10.10 with custom build of clang

```
➜  build git:(master) ✗ ../configure
...
configure: on OS X >=10.9, forcing use of clang
configure:
configure: CFG_ENABLE_CLANG     := 1
configure: CFG_USING_CLANG      := 1
configure: error: bad CLANG version: 3.7.0, need >=3.0svn
➜  build git:(master) ✗ clang -v
clang version 3.7.0 (trunk) (llvm/trunk 228281)
Target: x86_64-apple-darwin14.3.0
Thread model: posix
➜  build git:(master) ✗ which clang
/usr/local/bin/clang
➜  build git:(master) ✗ git log -1 | tee
commit c3d60aba6c86883c79055c1a3923d4db116b644e
Merge: 7044f4a 86a821e
Author: bors <bors@rust-lang.org>
Date:   Thu May 21 19:41:18 2015 +0000

    Auto merge of #25669 - GuillaumeGomez:typo-fix, r=Manishearth

    r? @steveklabnik
```

This is with clang built from source.

Guessing something in [here](https://github.com/rust-lang/rust/blob/master/configure#L978-L1004) is not right.

compare:

```
➜  build git:(master) ✗ /usr/bin/clang --version
Apple LLVM version 6.1.0 (clang-602.0.53) (based on LLVM 3.6.0svn)
Target: x86_64-apple-darwin14.3.0
Thread model: posix
➜  build git:(master) ✗ /usr/local/bin/clang --version
clang version 3.7.0 (trunk) (llvm/trunk 228281)
Target: x86_64-apple-darwin14.3.0
Thread model: posix
```

What was [that expression about regexes](http://xkcd.com/1171/)?

--- a/configure
+++ b/configure
@@ -986,7 +986,7 @@ then
             | cut -d ' ' -f 2)
 
         case $CFG_CLANG_VERSION in
-            (3.2* | 3.3* | 3.4* | 3.5* | 3.6*)
+            (3.2* | 3.3* | 3.4* | 3.5* | 3.6* | 3.7*)
             step_msg "found ok version of CLANG: $CFG_CLANG_VERSION"
             if [ -z "$CC" ]
             then

